### PR TITLE
Add a helper to change the text on various UI elements by platform

### PR DESCRIPTION
**Pull request by j-f1**
This adds a `menuTitle` API to automatically pick the correct title for each platform:

```ts
/** Pick the correct style for the menu title.
 * `darwinName` is autogenerated if not provided.
 */
menuTitle(winName: string, darwinName?: string): string
/** same as calling `menuTitle`, but it prepends “Show ” on macOS. */
menuTitle.show(winName: string): string
/** Translate a menu title from Windows style to macOS style */
menuTitle.winToDarwin(winName: string): string
/** Replace Windows-ese with macOS-ese (remove `&`, replace key terms) */
menuTitle.translate(winName: string): string
```
I chose to translate from Windows to macOS because Windows has extra features like `&` to signify shortcuts that I wanted to automatically remove from the macOS titles.

This also upgrades the npm shrinkwrap file to version 1 and moves the developer menu items to a submenu in the View submenu. If you’d rather keep those menu items where they are, I can revert that change.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,4561 @@
+{
```
Comment (j-f1) [inline package-lock.json]: Do you want me to remove this file? If so, I can do that.

Comment (shiftkey) [inline package-lock.json]: Please do. We're tracking the NPM 5 upgrade here #1806.

Comment (j-f1) [inline package-lock.json]: Gone.

**Comment by shiftkey** [comment]
@j-f1 unfortunately this sort of technical debt cleanup isn't something we're quite interested in at the moment - there's some bigger things we need to deal with right now, and we're a bit resource-constrained. And while it's not quite a user-facing enhancement like we talk about in [`How Do I Submit A (Good) Enhancement Suggestion?`](https://github.com/desktop/desktop/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion) in the `CONTRIBUTING.md` doc, we'd rather these sorts of changes start from an issue, so we can provide input early on and help guide things.

I think extracting a method only goes some of the way to addressing our long-term issues with how we currently do menus, so here's some suggestions that I have to think about:

 - can we write tests to catch regressions with the menu rules, conventions and behaviour we currently have? For example, we don't want to accidentally remove shortcuts without ensuring this is what we intend to do. I think we can do this without needing to extract a method, because `build-default-menu.ts` is decoupled from the UI.
 - how can we detect overlapping shortcuts with menu item names? [This comment](https://github.com/desktop/desktop/pull/2008#pullrequestreview-44573977) was the result of a human noticing it - are we able to automate these checks and not accidentally introduce a breakage?
 - these strings are all scattered around the codebase, meaning contributors need to do a find-and-replace if we have to change things - what would a more maintainable solution look like?

**Pull request closed without merging**